chapters/data-processing.tex

```diff
@@ -0,0 +1,783 @@
+and find out which variable or set of variables is the \textbf{unique identifier}\sidenote{\url{
+		https://dimewiki.worldbank.org/ID\_Variable\_Properties}} for each row.
+Ensuring that observations are uniquely and fully identified
+is arguably the most important step in data cleaning.
+It may be the case that the variables expected to be unique identifiers are either incomplete or contain duplicates.
+It is also possible for a data set to have no unique identifier, or that the identifier involves a long string, such as a name.
+In such cases, cleaning begins by carefully creating a variable that uniquely identifies the data.
+Note that while modern survey tools create unique identifiers for each submitted data record,
+that is not the same as having a unique ID variable for each individual in the sample.
+The unique identifier will be used to
+cross-referenced with other records, such as
+other rounds of data collection and
+different data sources.
+Therefore, you must keep a canonical dataset of all observations in your data.
+This \textbf{master dataset}\sidenote{\url{
+		https://dimewiki.worldbank.org/Master\_Data\_Set}}
+should contain a record of all units ever encountered for each relevant level of observation,
+and their corresponding unique identifiers.
+
+
+% Merchand
+Checking for duplicated entries and making sure that each observation has a unique identifier are usually part of data quality monitoring.
+The lack of a unique identifier should be addressed as soon as data is received,
+so you can keep track of observations.
+\texttt{ieduplicates} and \texttt{iecompdup},
+two Stata commands included in the \texttt{iefieldkit}
+package\index{iefieldkit},\sidenote{\url{https://dimewiki.worldbank.org/iefieldkit}}
+create an automated workflow to identify, correct and document
+occurrences of duplicate entries.
+One advantage of using \texttt{ieduplicates}\sidenote{\url{
+		https://dimewiki.worldbank.org/wiki/ieduplicates}} 
+to correct duplicated entries is that it creates \textit{duplicates reports}
+that record which corrections made and why.
+Even if you are not using this command,
+it is important to include similar records in the \texttt{Documentation} folder.
+
+\subsection{What makes data ``tidy''}
+
+% Where these concepts come from
+Before discussing the details of how to clean your data,
+we need to define a few concepts that will be used in this chapter.
+They refer to the structure of the data,
+and have long been discussed in the database management literature.
+This section is based on Hadley Wickham's definition of tidy data,\cite{tidy-data}
+which uses a terminology that is more intuitive for users of applied statistics,
+we have added our own twist based on the jargon of development research.
+
+% Definitions
+Development data is typically acquired in the form of \textbf{tables},
+two-dimensional datasets formed by rows and columns.\sidenote{\textbf{Table:}
```

Comment:
In this paragraph we are inconsistent with where we put the sidenote. I agree with what have been said before that in general it should come after the full sentence, but I think the exception to that should be when we define a word, like **table**, **data point**, **variable** and **tidy data**.

Reply:
fixed in 1029ad4

Reply:
The side note sits on the sentence in table.

Also, we should discuss repeating the definition from the text in _sidenote_. I am not opposed to it, but we should apply that consistently across the book if we want to keep that.